### ISSUE TITLE
Make sure details/summary is feature-complete before the next major release.

### ISSUE THREAD

**Issue by wolfbeast** [MoonchildProductions/Pale-Moon#1496]
With a recent change on github, the reactions panel with comments is now broken.
This component of github uses details/summary tags, which was added to our tree in #1041 but seems to have been incomplete or having some bugs. At the very least it looks like at least a number of the dependent bugs on [Mozilla's meta bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1245021) for this feature were not ported across, one of which is likely the cause for this problem. (my guess is [Bug 1299753](https://bugzilla.mozilla.org/show_bug.cgi?id=1299753) for this particular issue, but other deps should be looked at and if relevant, ported)

**Comment by wolfbeast** [comment]
My guess about the culprit was correct. @janekptacijarabaci you were the one starting on this feature and getting the base code in. Any reason why it was incomplete? Can you please check other dependent bugs on that meta for anything else that was missed?

**Comment by janekptacijarabaci** [comment]
> Any reason why it was incomplete?

I cannot track all METAs (I have limited time resources).

> Can you please check other dependent bugs on that meta for anything else that was missed?

I'm sorry, I'm currently dealing with other things...

**Comment by wolfbeast** [comment]
I'm sorry but I don't understand why you went about it this way.

If you want to implement a feature, that's fine, but either make sure you don't do it half-assed, or if you can't do it properly yourself in one go, that you make sure everyone knows (by indicating in the issue for the feature) what other outstanding issues there are so even if someone else would have to take over because of your time constraints, that things are clearly communicated!! I don't want us to become another half-assed-everything, piece-meal-implementation Mozilla. If we add a feature, we should add it as complete as possible.

Your list of bugs was _INCOMPLETE_. I checked the "maybe later" bugs you listed and they did not include what I tackled now; they were just bugs for keyboard access that are at most "nice to have" enhancements and not breaking -- if I had known there were serious functional bugs I wouldn't have had to spend 2 days working with github to figure out what their layout issue was with details/summary they were having in use, because I assumed it was feature-complete on par with what Mozilla ships in their releases now, and it would likely already have been fixed before it became a problem if I had known.
A bug in the implementation that you didn't mention, which was clearly indicated on Mozilla's meta-bug that I assume you used as a guideline to implement what you did. The least you could have done was point out open issues you didn't fix yet. :-1: 

So, for the record then, dependent bugs for this feature:

Depends on: | 1221416 1245032 1245047 1245395 1245430 1245431 1258948 1269642 1303906 1307679 1308080 1382124 591737 634004 1225412 1225752 1245036 1245424 1246185 1246404 1258657 1264533 1271549 1299753 1304441 1321166

If anyone has free time, please do a cursory check of these bugs and indicate which have already been taken care of, are not applicable, or have already been implemented by @janekptacijarabaci or myself, so we don't get any more nasty surprises when something on the web just breaks inexplicably.

**Comment by janekptacijarabaci** [comment]
Ad META: I did not take it into account - then. Otherwise, this: [bug 1249556](https://bugzilla.mozilla.org/show_bug.cgi?id=1249556) is also missing (and I do not know why).

Yes, it was not all that. I haven't found all "the connections" (not intentionally, of course). I will not create new tags/elements - anymore (if by chance) - it will be better for both sides (someone else will definitely be more successful)...

And I think here we differ in our opinion of what `serious functional bugs` is. But this is the way the world goes...

**Comment by wolfbeast** [comment]
1249556 is under "ship-details-summary" on bugzilla (because it's an enhancement that is peripheral to the feature, which Mozilla wanted to have before switching it on for everyone) which is most certainly linked to the overall meta. dependencies go more than 1 level deep, you know... Or maybe you don't.

I'm also not asking you to stop offering enhancements or new functionality... I'm just asking you to make sure things are reasonably **complete** and **with all known bugs fixed** when you offer such things, and if that's too much work, to **communicate** any open issues with the feature so I know when to merge things in (i.e.: when the feature is complete and ready to be exposed to the public, not when it's only half-done). But if that's too much work and doesn't fit in your busy schedule, then by all means, take things down a notch or two.

If your code messes up the layout of a page, then that is a `serious functional bug`. Don't tell me you disagree with that. 
![image1](https://user-images.githubusercontent.com/3359132/33296672-12ea67d8-d3dd-11e7-8691-1c49283f548e.jpg)

**Comment by janekptacijarabaci** [comment]
> I'm just asking you to make sure things are reasonably complete

Once again: it was not intentional.

> Doesn't fit in your busy schedule
> If your code messes up the layout of a page, then that is a serious functional bug. Don't tell me you disagree with that.

Ad `I cannot track all METAs (I have limited time resources)`. Yes, I do not watch this META. There may remain some uncorrected bugs (maybe a long time - and it can still grow bigger), e.g.: [bug 1307679](https://bugzilla.mozilla.org/show_bug.cgi?id=1307679). And the design will be different, also. And surely there must be someone out there who would welcome, also.
Like many other things...

**Comment by janekptacijarabaci** [comment]
The following things (I'll look later today):
https://bugzilla.mozilla.org/show_bug.cgi?id=1245036
https://bugzilla.mozilla.org/show_bug.cgi?id=1264533 (partially)
https://bugzilla.mozilla.org/show_bug.cgi?id=1304441
